scripts/apps/extension-points/directives/ExtensionPointDirective.js

```diff
@@ -15,26 +15,18 @@ import React from 'react';
  */
 ExtensionPointDirective.$inject = ['extensionPoints'];
 export function ExtensionPointDirective(extensionPoints) {
-    function propsFromData(data, scope) {
-        var json = {};
-
-        _.forEach(data, (value) => {
-            json[value] = scope.$parent.$eval(value);
+    function _buildCompoment(extension, scope) {
+        // for easy access put values from parent scope into the redux store of the extension
+        _.forEach(extension.data, (value) => {
+            extension.props.store.getState()[value] = scope.$parent.$eval(value);
         });
-        return json;
-    }
-
-    function buildCompoment(extension, scope) {
-        return React.createElement(
-            extension.componentClass,
-            propsFromData(extension.data, scope)
-        );
+        return React.createElement(extension.componentClass, extension.props);
     }
 
     return {
         link: function(scope, elem, attr) {
-            var registeredExtenstions = extensionPoints[attr.sdExtensionPoint];
-            var components = _.map(registeredExtenstions, (extension) => buildCompoment(extension, scope));
+            var registeredExtenstions = extensionPoints.extensions[attr.sdExtensionPoint];
```

Comment:
- I think you mean `registeredExtension` here (the perils of auto-complete 😄 ). Additionally, it seems a bit unorthodox to me to access this property directly on the service. Since we already have a `register` method, why not also a getter (ie. `get`) ?

- Let's not assume this value exists in the object and check for it (and perhaps `throw` if not). This could be checked in the method suggested in my previous point.

Reply:
I'm adding a check now, but it will return an empty array if there aren't any registered extension for that specific extension point. It is not an error if not all of the possible extension points are actually used.